File: week1/cpp/dilate.cpp
/*
Copyright 2017 BIG VISION LLC ALL RIGHTS RESERVED

This code is made available to the students of 
the online course titled "Computer Vision for Faces" 
by Satya Mallick for personal non-commercial use. 

Sharing this code is strictly prohibited without written
permission from Big Vision LLC. 

For licensing and other inquiries, please email 
spmallick@bigvisionllc.com 
*/

#include <iostream>
#include <opencv2/highgui.hpp>
#include <opencv2/imgproc.hpp>

using namespace cv;
using namespace std;

int main() {
    Mat image;
    // Image read in and stored in Mat image
    image = imread("../data/images/truth.png", IMREAD_COLOR);
    
    // Create a structuring element
    int dilationSize = 6;
    Mat element = getStructuringElement(cv::MORPH_CROSS,
           cv::Size(2 * dilationSize + 1, 2 * dilationSize + 1),
           cv::Point(dilationSize, dilationSize));

    // Dilated image is stored in Mat imageDilated
    Mat imageDilated;

    // Dilation will increase brightness.First parameter is the original image, second is the dilated image
    dilate(image, imageDilated, element);

    // Display original image
    namedWindow("Original Image", CV_WINDOW_AUTOSIZE);
    imshow("Original Image", image);  
   
    // Display dilated image
    namedWindow("Dilation", CV_WINDOW_AUTOSIZE);
    imshow("Dilation", imageDilated);  

    // Wait for the user to press any key
    waitKey(0);
    return 0;
}

File: week1/python/dilate.py
# Copyright 2017 BIG VISION LLC ALL RIGHTS RESERVED
# 
# This code is made available to the students of 
# the online course titled "Computer Vision for Faces" 
# by Satya Mallick for personal non-commercial use. 
#
# Sharing this code is strictly prohibited without written
# permission from Big Vision LLC. 
#
# For licensing and other inquiries, please email 
# spmallick@bigvisionllc.com 
#

import cv2

imageName = "../data/images/truth.png"

# Read the input image
image = cv2.imread(imageName, cv2.IMREAD_COLOR)

# Check for an invalid input
if image is None:  
    print("Could not open or find the image")

# Get structuring element/kernel which will be used for dilation
dilationSize = 6
element = cv2.getStructuringElement(cv2.MORPH_CROSS, (2*dilationSize+1, 2*dilationSize+1),
                                    (dilationSize, dilationSize))

# Apply dilate function on the input image
imageDilated = cv2.dilate(image, element)    

# Display the original image
cv2.namedWindow("Original Image", cv2.WINDOW_AUTOSIZE)
cv2.imshow("Original Image", image)		

# Display the dilated image
cv2.namedWindow("Dilation", cv2.WINDOW_AUTOSIZE)
cv2.imshow("Dilation", imageDilated)
	
# Wait for user to press any key
cv2.waitKey(0)
cv2.destroyAllWindows()
